details.js

```diff
@@ -0,0 +1,88 @@
+document.querySelector('.loader').style.display = 'none';
+
+    const urlParams = new URLSearchParams(window.location.search);
+    const countryName = urlParams.get('country');
+
+if (countryName) {
+    fetch('./CountriesData.json')
+        .then(response => response.json())
+        .then(data => {
+        const countryDetails = document.querySelector('.country-details');
+        countryDetails.classList.add('country-details');
+
+        const selectedCountry = data.find(country => country.name === countryName);
+        console.log('selectedCountry:', selectedCountry);
+        
+        const countryInfo = document.createElement('div');
+        countryInfo.classList.add('country-info');
+
+        const selectedCountryName = document.createElement('h1');
+        selectedCountryName.innerHTML = selectedCountry.name;
+        selectedCountryName.classList.add('h1');
+
+        
+        const countryDataList = document.createElement('ul');
+        countryDataList.classList.add('country-data-list');
+
+        const countryNativeName = document.createElement('li');
+        countryNativeName.innerHTML = `<strong>Native Name:</strong> ${selectedCountry.name}`; ;
+        console.log('selectedCountry.name:', countryNativeName);
+
+        const countryPopulation = document.createElement('li');
+        countryPopulation.innerHTML = `<strong>Population:</strong> ${selectedCountry.population}`;
+
```

Comment:
You can extrac this into a function and call it inside a loop to prevent code duplication